byway/src/main/webapp/routepage.js

```diff
@@ -292,5 +292,45 @@ function createRecButton(rec){
   return recBtn;
 }
 
-/* exported initMap, generateRoute, placesService, map */
-/* global google, findPlace, getTripKeyFromUrl, configureTripKeyForPath */
\ No newline at end of file
+
+/** Creates a URL link to Google Maps based on the start/end point and route
+ *  @returns {String} routeLink url containing query params for the user’s route
+ */
+function generateRouteLink(){
+    let routeLink = "https://www.google.com/maps/dir/?api=1&travelmode=driving" 
+    routeLink += "&origin=" + start.name.split(' ').join('%20') + "&destination=" + 
+      end.name.split(' ').join('%20') + "&waypoints=";
+    for(let place of route){
+      routeLink += place.name.split(' ').join('%20') + "|";
+    }
+   return routeLink;
+}
+
+/** Determines if email address is valid using regex patterns
+ *  @param {String} email email address 
+ *  @returns {boolean} is the email a valid email address
+ */
+function validateEmail(email) {
+    const re = /^\w+([-]?\w+)*@\w+([-]?\w+)*(\.\w{2,3})+$/;
+    return re.test(String(email).toLowerCase());
+}
+
+/** Opens a new window of an email client with prefilled to, subject, and body field 
+ *  containing a Google Maps link of the user's route 
+ */
+function sendEmail(){
+  let emailAddress = document.getElementById("email-address").value;
+  let emailLink = "mailto:" + emailAddress + "?subject=Your%20Roadtrip%20Plan&body=Thank%20you%20for%20using%20Byway!%20" +
```

Comment:
Please don't write manually encoded values like this, too easy to mess up.

Also, since these are query parameters again, you can use `URLSearchParams` here too.

Reply:
Changed to use URLSeachParams()